aws/scripts/py/cache_invalidation.py

```
@@ -0,0 +1,302 @@
+            # Process staging first, then production
+            environments_to_process = [Environment.STAGING, Environment.PRODUCTION]
+            
+            for step, environment in enumerate(environments_to_process, 1):
+                self.logger.info(f"Step {step}/{len(environments_to_process)}: {environment.value}")
+                distribution = distributions[environment]
+                self._invalidate_environment(distribution, environment)
+
+            self.logger.info("Blue-green cache invalidation completed successfully")
+
+        except Exception as e:
+            self.logger.error(f"Cache invalidation failed: {e}")
```

Ruff (0.11.9)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)